src/test/java/com/solvd/noteservice/NoteServiceTest.java

```diff
@@ -0,0 +1,168 @@
+package com.solvd.noteservice;
+
+import com.solvd.noteservice.domain.Note;
+import com.solvd.noteservice.domain.exception.IllegalOperationException;
+import com.solvd.noteservice.domain.exception.ResourceDoesNotExistException;
+import com.solvd.noteservice.kafka.KfProducer;
+import com.solvd.noteservice.repository.NoteRepository;
+import com.solvd.noteservice.service.NoteService;
+import com.solvd.noteservice.service.client.ElasticClient;
+import com.solvd.noteservice.service.impl.NoteServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.doAnswer;
+
+@ExtendWith(MockitoExtension.class)
+public class NoteServiceTest {
+
+    private NoteService noteService;
+
+    @Mock
+    private NoteRepository noteRepository;
+
+    @Mock
+    private ElasticClient elasticClient;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private KfProducer kfProducer;
+
+    @BeforeEach
+    void setUp(){
+        noteService = new NoteServiceImpl(noteRepository,
+                elasticClient, restTemplate, kfProducer);
+    }
+
+    @Test
+    void verifyExistsByIdPassedTest() {
+        Long noteId = 1L;
+        when(elasticClient.isExistById(anyLong())).thenReturn(false);
+        boolean result = noteService.isExistById(noteId);
+        assertEquals(false, result, "Assert that method returns false");
+        verify(elasticClient, times(1)).isExistById(noteId);
+    }
+
+    @Test
+    void verifyCreatePassedTest() {
+        Note note = NoteFactory.generateNote();
+        Long noteId = 1L;
+        note.setId(null);
+        when(restTemplate.getForObject(anyString(), any(), anyLong())).thenReturn(true);
+        doAnswer(invocation -> {
+            Note noteCreated = invocation.getArgument(0);
+            noteCreated.setId(noteId);
+            return noteCreated;
+        }).when(noteRepository).save(note);
+        note = noteService.create(note);
+        assertEquals(noteId, note.getId(), "Assert that notes id are equals");
+        verify(restTemplate, times(1)).getForObject(anyString(), any(), anyLong());
+        verify(noteRepository, times(1)).save(note);
```

Comment:
@vpetrovaa this test has only 2 lines with assertions
other lines just preparation for those assertions, the just the complexity, making such test great candidate for deleting, when 
developers will face problems with it

so, exclude this configuration from the test, your tests should have only assertions

Reply:
@h1alexbel fixed https://github.com/vpetrovaa/note-service/pull/18/commits/788e4b539353756e0dcbce51360a5c47da127479